packages/azure/src/lib/ConsumptionManagement.ts

```diff
@@ -116,11 +116,13 @@ export default class ConsumptionManagementService {
     const unknownRows: ConsumptionDetailRow[] = []
 
     usageRows
-      .filter(
-        (consumptionRow) =>
-          new Date(consumptionRow.date || Date.now()) >= startDate &&
-          new Date(consumptionRow.date || Date.now()) <= endDate,
-      )
+      .filter((consumptionRow) => {
+        const consumptionRowDate = consumptionRow.date || Date.now()
```

Comment:
Have a default value of the current date (`Date.now()`) may introduce a bug where the row is still included if the user has an endDate that matches the current date. It might be better to have an if-statement that checks if the date property is present, and if it isn't, it returns false (excluding that row)